frontend/src/metabase/visualizations/components/settings/ColumnItem.jsx

```diff
@@ -28,19 +34,33 @@ const ColumnItem = ({
   onEdit,
   onEnable,
   draggable,
-}) => (
-  <ColumnItemRoot draggable={draggable} onClick={onClick}>
-    <ColumnItemContainer>
-      {draggable && <ColumnItemDragHandle name="grabber2" size={12} />}
-      <ColumnItemContent>
-        <ColumnItemSpan>{title}</ColumnItemSpan>
-        {onEdit && <ActionIcon icon="ellipsis" onClick={onEdit} />}
-        {onAdd && <ActionIcon icon="add" onClick={onAdd} />}
-        {onRemove && <ActionIcon icon="eye_filled" onClick={onRemove} />}
-        {onEnable && <ActionIcon icon="eye_crossed_out" onClick={onEnable} />}
-      </ColumnItemContent>
-    </ColumnItemContainer>
-  </ColumnItemRoot>
-);
+}) => {
+  const settingsRef = useRef(null);
+
+  const handleOnEdit = () => {
+    onEdit(settingsRef.current);
```

Comment:
I think we could pass an event from `ActionIcon` so that we don't even need to forward its ref. And then pass `event.target` instead.